Add data retention and upgrade safety advice (#4440)
vaadin/docs: articles/control-center/getting-started/upgrade.adoc

    --reuse-values
----

> .Securing your data

Reviewer: ```suggestion
.Securing Your Data
```

[TIP]

> Although nothing should happen to Control Center-managed databases and users when upgrading, it is advisable to perform a xref:../database/backups.adoc[Backup] before proceeding with an upgrade to prevent any data loss.

Reviewer: ```suggestion
Although nothing should happen to Control Center-managed databases and users when upgrading, you should perform a <<../database/backups#,Backup>> before proceeding with an upgrade to prevent any data loss.
```
